src/cli.c

```diff
@@ -28,153 +28,228 @@
-       the script */
-    optind = 1;
-    while (optind < argc && *argv[optind] == '-') {
+    /* cannot use getopt because we want to pass the command line to the script */
+    int optind = 1;
+    while (optind < argc && *argv[optind] == OPT_PREFIX) {
         char *arg = argv[optind] + 1;
-        const char *longopt = "";
-        /* a single - is not an option, it also stops argument scanning */
-        if (!*arg)
+        CLIOption opt = { .key = 0, .name = NULL, .length = 0 };
+        if (!get_option(&arg, &opt))
             break;
-        optind++;
-        if (*arg == '-') {
-            longopt = arg + 1;
-            arg += strlen(arg);
-            /* -- stops argument scanning */
-            if (!*longopt)
-                break;
+        optind += 1;
+        /* combining short options is NOT supported */
+        if (opt.key && opt.length > 0) {
+            report_bad_option(arg - 1);
+            exit_code = EXIT_INVALID_ARG;
+            goto exit;
         }
-        for (; *arg || *longopt; longopt = "") {
-            char opt = *arg;
-            if (opt)
-                arg++;
-            if (opt == 'h' || opt == '?' || !strcmp(longopt, "help")) {
-                help();
-                continue;
+        while (opt.key || *opt.name) {
+            if (opt.key == 'v' || is_longopt(opt, "version")) {
+                print_version();
+                goto exit;
             }
-            if (opt == 'e' || !strcmp(longopt, "eval")) {
-                if (*arg) {
-                    expr = arg;
-                    break;
-                }
-                if (optind < argc) {
-                    expr = argv[optind++];
-                    break;
-                }
-                fprintf(stderr, "tjs: missing expression for -e\n");
-                exit(2);
+            if (opt.key == 'h' || is_longopt(opt, "help")) {
+                print_help();
+                goto exit;
             }
-            if (!strcmp(longopt, "override-filename")) {
-                if (*arg) {
-                    override_filename = arg;
-                    break;
-                }
-                if (optind < argc) {
-                    override_filename = argv[optind++];
+            if (opt.key == 'e' || is_longopt(opt, "eval")) {
+                flags.eval_expr = get_option_value(arg, argc, argv, &optind);
+                if (flags.eval_expr)
                     break;
-                }
-                fprintf(stderr, "tjs: missing expression for --override-filename\n");
-                exit(2);
+                report_missing_argument(&opt);
+                exit_code = EXIT_INVALID_ARG;
+                goto exit;
             }
-            if (opt == 'i' || !strcmp(longopt, "interactive")) {
-                interactive++;
-                continue;
+            if (is_longopt(opt, "override-filename") || is_longopt(opt, "overrideFilename")) {
```

Comment:
why the extra spelling? let's just have `override-filename`

Reply:
Just for sake of convenience. That weird pascal casing arg style got popular after yarn introduced it. My thinking was it is easy enough to implement and makes everybody feel at home regardless of their background/expectations.

Reply:
I'd rather not have it :-)

Reply:
Ok, I'll remove it.